logstash-core/src/main/java/org/logstash/config/ir/compiler/VariableDefinition.java

```diff
@@ -72,6 +72,8 @@ public String generateCode() {
             safe = EventCondition.class;
         } else if (DynamicMethod.class.isAssignableFrom(clazz)) {
             safe = DynamicMethod.class;
+        } else if (BaseDataset.class.isAssignableFrom(clazz)) {
```

Comment:
the simplicity and impact of this change is simply

![FirsthandUnknownGoitered-small](https://user-images.githubusercontent.com/31809/85536443-03a10080-b60b-11ea-872d-e5ca5de26975.gif)

Reply:
We may want to look at whether this can resolve to the `Dataset` interface (as opposed to the `BaseDataset` abstract class that implements `Dataset` with a few other methods).

Reply:
With the three pipelines I used to benchmark (quite big: 1.5k, 5k and 7k lines):
baseline 7.6.2: 999 classes
regresssion: 1320 (essentially all)
with this PR + #12038: 40 classes
with this PR + #12038 + changing to Dataset instead of BaseDataset: 21 classes